From ca573f42e31400b1e8301e75a74424a9248a1f75 Mon Sep 17 00:00:00 2001
From: Travis Geiselbrecht <travisg@gmail.com>
Date: Tue, 11 Jun 2019 15:37:00 -0700
Subject: [PATCH] reapply the systick patch to cmsis

SysTick_Config() tries to set the systick priority to max. Comment out
this line since arch/arm/arm-m/arch.c sets the priority to medium to
compete with the rest of the exceptions in the system.

Also include lk/compiler.h to avoid redundant cmsis macros.
---
 external/arch/arm/arm-m/CMSIS/Include/cmsis_gcc.h      | 3 +++
 external/arch/arm/arm-m/CMSIS/Include/core_armv81mml.h | 2 +-
 external/arch/arm/arm-m/CMSIS/Include/core_armv8mbl.h  | 2 +-
 external/arch/arm/arm-m/CMSIS/Include/core_armv8mml.h  | 2 +-
 external/arch/arm/arm-m/CMSIS/Include/core_cm0.h       | 2 +-
 external/arch/arm/arm-m/CMSIS/Include/core_cm0plus.h   | 2 +-
 external/arch/arm/arm-m/CMSIS/Include/core_cm1.h       | 2 +-
 external/arch/arm/arm-m/CMSIS/Include/core_cm23.h      | 2 +-
 external/arch/arm/arm-m/CMSIS/Include/core_cm3.h       | 2 +-
 external/arch/arm/arm-m/CMSIS/Include/core_cm33.h      | 2 +-
 external/arch/arm/arm-m/CMSIS/Include/core_cm35p.h     | 2 +-
 external/arch/arm/arm-m/CMSIS/Include/core_cm4.h       | 2 +-
 external/arch/arm/arm-m/CMSIS/Include/core_cm55.h      | 2 +-
 external/arch/arm/arm-m/CMSIS/Include/core_cm7.h       | 2 +-
 external/arch/arm/arm-m/CMSIS/Include/core_sc000.h     | 2 +-
 external/arch/arm/arm-m/CMSIS/Include/core_sc300.h     | 2 +-
 16 files changed, 18 insertions(+), 15 deletions(-)

diff --git a/external/arch/arm/arm-m/CMSIS/Include/cmsis_gcc.h b/external/arch/arm/arm-m/CMSIS/Include/cmsis_gcc.h
index 67bda4ef..dc3ae357 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/cmsis_gcc.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/cmsis_gcc.h
@@ -25,6 +25,9 @@
 #ifndef __CMSIS_GCC_H
 #define __CMSIS_GCC_H
 
+/* LK: include lk's compiler.h first, which has some of the same #defines */
+#include <lk/compiler.h>
+
 /* ignore some GCC warnings */
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wsign-conversion"
diff --git a/external/arch/arm/arm-m/CMSIS/Include/core_armv81mml.h b/external/arch/arm/arm-m/CMSIS/Include/core_armv81mml.h
index 4a15992f..7e95a520 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/core_armv81mml.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/core_armv81mml.h
@@ -4084,7 +4084,7 @@ __STATIC_INLINE uint32_t SysTick_Config(uint32_t ticks)
   }
 
   SysTick->LOAD  = (uint32_t)(ticks - 1UL);                         /* set reload register */
-  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
+// LK  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
   SysTick->VAL   = 0UL;                                             /* Load the SysTick Counter Value */
   SysTick->CTRL  = SysTick_CTRL_CLKSOURCE_Msk |
                    SysTick_CTRL_TICKINT_Msk   |
diff --git a/external/arch/arm/arm-m/CMSIS/Include/core_armv8mbl.h b/external/arch/arm/arm-m/CMSIS/Include/core_armv8mbl.h
index 932d3d18..eeb0342e 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/core_armv8mbl.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/core_armv8mbl.h
@@ -2168,7 +2168,7 @@ __STATIC_INLINE uint32_t SysTick_Config(uint32_t ticks)
   }
 
   SysTick->LOAD  = (uint32_t)(ticks - 1UL);                         /* set reload register */
-  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
+// LK  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
   SysTick->VAL   = 0UL;                                             /* Load the SysTick Counter Value */
   SysTick->CTRL  = SysTick_CTRL_CLKSOURCE_Msk |
                    SysTick_CTRL_TICKINT_Msk   |
diff --git a/external/arch/arm/arm-m/CMSIS/Include/core_armv8mml.h b/external/arch/arm/arm-m/CMSIS/Include/core_armv8mml.h
index 2bd9e760..80bd6818 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/core_armv8mml.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/core_armv8mml.h
@@ -3065,7 +3065,7 @@ __STATIC_INLINE uint32_t SysTick_Config(uint32_t ticks)
   }
 
   SysTick->LOAD  = (uint32_t)(ticks - 1UL);                         /* set reload register */
-  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
+// LK NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
   SysTick->VAL   = 0UL;                                             /* Load the SysTick Counter Value */
   SysTick->CTRL  = SysTick_CTRL_CLKSOURCE_Msk |
                    SysTick_CTRL_TICKINT_Msk   |
diff --git a/external/arch/arm/arm-m/CMSIS/Include/core_cm0.h b/external/arch/arm/arm-m/CMSIS/Include/core_cm0.h
index 6441ff34..f4d5d7e9 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/core_cm0.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/core_cm0.h
@@ -928,7 +928,7 @@ __STATIC_INLINE uint32_t SysTick_Config(uint32_t ticks)
   }
 
   SysTick->LOAD  = (uint32_t)(ticks - 1UL);                         /* set reload register */
-  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
+// LK  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
   SysTick->VAL   = 0UL;                                             /* Load the SysTick Counter Value */
   SysTick->CTRL  = SysTick_CTRL_CLKSOURCE_Msk |
                    SysTick_CTRL_TICKINT_Msk   |
diff --git a/external/arch/arm/arm-m/CMSIS/Include/core_cm0plus.h b/external/arch/arm/arm-m/CMSIS/Include/core_cm0plus.h
index 4e7179a6..7c589dc1 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/core_cm0plus.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/core_cm0plus.h
@@ -1063,7 +1063,7 @@ __STATIC_INLINE uint32_t SysTick_Config(uint32_t ticks)
   }
 
   SysTick->LOAD  = (uint32_t)(ticks - 1UL);                         /* set reload register */
-  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
+// LK  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
   SysTick->VAL   = 0UL;                                             /* Load the SysTick Counter Value */
   SysTick->CTRL  = SysTick_CTRL_CLKSOURCE_Msk |
                    SysTick_CTRL_TICKINT_Msk   |
diff --git a/external/arch/arm/arm-m/CMSIS/Include/core_cm1.h b/external/arch/arm/arm-m/CMSIS/Include/core_cm1.h
index 76b45697..aebd6960 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/core_cm1.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/core_cm1.h
@@ -955,7 +955,7 @@ __STATIC_INLINE uint32_t SysTick_Config(uint32_t ticks)
   }
 
   SysTick->LOAD  = (uint32_t)(ticks - 1UL);                         /* set reload register */
-  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
+// LK  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
   SysTick->VAL   = 0UL;                                             /* Load the SysTick Counter Value */
   SysTick->CTRL  = SysTick_CTRL_CLKSOURCE_Msk |
                    SysTick_CTRL_TICKINT_Msk   |
diff --git a/external/arch/arm/arm-m/CMSIS/Include/core_cm23.h b/external/arch/arm/arm-m/CMSIS/Include/core_cm23.h
index 55fff995..717bc828 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/core_cm23.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/core_cm23.h
@@ -2243,7 +2243,7 @@ __STATIC_INLINE uint32_t SysTick_Config(uint32_t ticks)
   }
 
   SysTick->LOAD  = (uint32_t)(ticks - 1UL);                         /* set reload register */
-  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
+// LK  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
   SysTick->VAL   = 0UL;                                             /* Load the SysTick Counter Value */
   SysTick->CTRL  = SysTick_CTRL_CLKSOURCE_Msk |
                    SysTick_CTRL_TICKINT_Msk   |
diff --git a/external/arch/arm/arm-m/CMSIS/Include/core_cm3.h b/external/arch/arm/arm-m/CMSIS/Include/core_cm3.h
index 74fb87e5..fd2bcfc8 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/core_cm3.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/core_cm3.h
@@ -1841,7 +1841,7 @@ __STATIC_INLINE uint32_t SysTick_Config(uint32_t ticks)
   }
 
   SysTick->LOAD  = (uint32_t)(ticks - 1UL);                         /* set reload register */
-  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
+// LK  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
   SysTick->VAL   = 0UL;                                             /* Load the SysTick Counter Value */
   SysTick->CTRL  = SysTick_CTRL_CLKSOURCE_Msk |
                    SysTick_CTRL_TICKINT_Msk   |
diff --git a/external/arch/arm/arm-m/CMSIS/Include/core_cm33.h b/external/arch/arm/arm-m/CMSIS/Include/core_cm33.h
index f9cf6ab1..e0ba8e0e 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/core_cm33.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/core_cm33.h
@@ -3133,7 +3133,7 @@ __STATIC_INLINE uint32_t SysTick_Config(uint32_t ticks)
   }
 
   SysTick->LOAD  = (uint32_t)(ticks - 1UL);                         /* set reload register */
-  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
+// LK  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
   SysTick->VAL   = 0UL;                                             /* Load the SysTick Counter Value */
   SysTick->CTRL  = SysTick_CTRL_CLKSOURCE_Msk |
                    SysTick_CTRL_TICKINT_Msk   |
diff --git a/external/arch/arm/arm-m/CMSIS/Include/core_cm35p.h b/external/arch/arm/arm-m/CMSIS/Include/core_cm35p.h
index 552c2946..ab65cc51 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/core_cm35p.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/core_cm35p.h
@@ -3133,7 +3133,7 @@ __STATIC_INLINE uint32_t SysTick_Config(uint32_t ticks)
   }
 
   SysTick->LOAD  = (uint32_t)(ticks - 1UL);                         /* set reload register */
-  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
+// LK  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
   SysTick->VAL   = 0UL;                                             /* Load the SysTick Counter Value */
   SysTick->CTRL  = SysTick_CTRL_CLKSOURCE_Msk |
                    SysTick_CTRL_TICKINT_Msk   |
diff --git a/external/arch/arm/arm-m/CMSIS/Include/core_cm4.h b/external/arch/arm/arm-m/CMSIS/Include/core_cm4.h
index e21cd149..52ef4922 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/core_cm4.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/core_cm4.h
@@ -2027,7 +2027,7 @@ __STATIC_INLINE uint32_t SysTick_Config(uint32_t ticks)
   }
 
   SysTick->LOAD  = (uint32_t)(ticks - 1UL);                         /* set reload register */
-  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
+// LK  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
   SysTick->VAL   = 0UL;                                             /* Load the SysTick Counter Value */
   SysTick->CTRL  = SysTick_CTRL_CLKSOURCE_Msk |
                    SysTick_CTRL_TICKINT_Msk   |
diff --git a/external/arch/arm/arm-m/CMSIS/Include/core_cm55.h b/external/arch/arm/arm-m/CMSIS/Include/core_cm55.h
index 51b68e6d..e0af1db5 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/core_cm55.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/core_cm55.h
@@ -4145,7 +4145,7 @@ __STATIC_INLINE uint32_t SysTick_Config(uint32_t ticks)
   }
 
   SysTick->LOAD  = (uint32_t)(ticks - 1UL);                         /* set reload register */
-  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
+// LK NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
   SysTick->VAL   = 0UL;                                             /* Load the SysTick Counter Value */
   SysTick->CTRL  = SysTick_CTRL_CLKSOURCE_Msk |
                    SysTick_CTRL_TICKINT_Msk   |
diff --git a/external/arch/arm/arm-m/CMSIS/Include/core_cm7.h b/external/arch/arm/arm-m/CMSIS/Include/core_cm7.h
index 010506e9..3a45a63b 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/core_cm7.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/core_cm7.h
@@ -2264,7 +2264,7 @@ __STATIC_INLINE uint32_t SysTick_Config(uint32_t ticks)
   }
 
   SysTick->LOAD  = (uint32_t)(ticks - 1UL);                         /* set reload register */
-  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
+// LK  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
   SysTick->VAL   = 0UL;                                             /* Load the SysTick Counter Value */
   SysTick->CTRL  = SysTick_CTRL_CLKSOURCE_Msk |
                    SysTick_CTRL_TICKINT_Msk   |
diff --git a/external/arch/arm/arm-m/CMSIS/Include/core_sc000.h b/external/arch/arm/arm-m/CMSIS/Include/core_sc000.h
index dbc755ff..fe52f94c 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/core_sc000.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/core_sc000.h
@@ -1006,7 +1006,7 @@ __STATIC_INLINE uint32_t SysTick_Config(uint32_t ticks)
   }
 
   SysTick->LOAD  = (uint32_t)(ticks - 1UL);                         /* set reload register */
-  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
+// LK  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
   SysTick->VAL   = 0UL;                                             /* Load the SysTick Counter Value */
   SysTick->CTRL  = SysTick_CTRL_CLKSOURCE_Msk |
                    SysTick_CTRL_TICKINT_Msk   |
diff --git a/external/arch/arm/arm-m/CMSIS/Include/core_sc300.h b/external/arch/arm/arm-m/CMSIS/Include/core_sc300.h
index d6662103..ce1405f9 100644
--- a/external/arch/arm/arm-m/CMSIS/Include/core_sc300.h
+++ b/external/arch/arm/arm-m/CMSIS/Include/core_sc300.h
@@ -1815,7 +1815,7 @@ __STATIC_INLINE uint32_t SysTick_Config(uint32_t ticks)
   }
 
   SysTick->LOAD  = (uint32_t)(ticks - 1UL);                         /* set reload register */
-  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
+// LK  NVIC_SetPriority (SysTick_IRQn, (1UL << __NVIC_PRIO_BITS) - 1UL); /* set Priority for Systick Interrupt */
   SysTick->VAL   = 0UL;                                             /* Load the SysTick Counter Value */
   SysTick->CTRL  = SysTick_CTRL_CLKSOURCE_Msk |
                    SysTick_CTRL_TICKINT_Msk   |
-- 
2.25.1

